lonboard/geopandas.py

```diff
@@ -0,0 +1,310 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+from numpy import uint8
+
+from lonboard import Map, viz
+from lonboard.basemap import CartoBasemap
+from lonboard.colormap import apply_categorical_cmap, apply_continuous_cmap
+
+if TYPE_CHECKING:
+    from numpy.typing import ArrayLike, NDArray
+    from pandas.core.series import Series
+
+    from lonboard.types.layer import (
+        IntFloat,
+        PathLayerKwargs,
+        PolygonLayerKwargs,
+        ScatterplotLayerKwargs,
+    )
+    from lonboard.types.map import MapKwargs
+
+__all__ = ["LonboardAccessor"]
+
+_QUERY_NAME_TRANSLATION = str.maketrans(dict.fromkeys("., -_/", ""))
+_basemap_providers = {
+    "CartoDB Positron": CartoBasemap.Positron,
+    "CartoDB Positron No Label": CartoBasemap.PositronNoLabels,
+    "CartoDB Darkmatter": CartoBasemap.DarkMatter,
+    "CartoDB Darkmatter No Label": CartoBasemap.DarkMatterNoLabels,
+    "CartoDB Voyager": CartoBasemap.Voyager,
+    "CartoDB Voyager No Label": CartoBasemap.VoyagerNoLabels,
+}
+# Convert keys to lower case without spaces
+_BASEMAP_PROVIDERS = {
+    k.translate(_QUERY_NAME_TRANSLATION).lower(): v
+    for k, v in _basemap_providers.items()
+}
+
+
+@pd.api.extensions.register_dataframe_accessor("lb")
+class LonboardAccessor:
+    """Geopandas Extension class to provide the `explore` method."""
+
+    def __init__(self, pandas_obj: gpd.GeoDataFrame) -> None:
+        """Initialize geopandas extension."""
+        self._validate(pandas_obj)
+        self._obj = pandas_obj
+
+    @staticmethod
+    def _validate(obj: gpd.GeoDataFrame) -> None:
+        if not isinstance(obj, gpd.GeoDataFrame):
+            raise TypeError("Input Must be a geodataframe")
+
+    def explore(  # noqa: C901, PLR0912, PLR0913, PLR0915
+        self,
+        *,
```

Comment:
I know this is the opposite of what I [said before](https://github.com/developmentseed/lonboard/pull/728#discussion_r2153144389), but I think we should remove `*` so that the signature is exactly the same as the upstream `GeoDataFrame.explore` (even though the thought of someone passing in all these parameters positionally is 😬

Reply:
I'd like this to be in line with future GeoPandas, so if GeoPandas is likely to add in `*`, then we might as well do it here

Reply:
Ok, let's keep the `*` here after `column`

Reply:
ok, I'll still update the arguments to put them in the same order (just because) and make sure the default values are consistent. My preference would be to stick lonboard-specific stuff like wireframe and elevation at the end of the signature (assuming they'll continue to pass through to the existing explore's **kwargs), though I can remove them entirely if that's still your preference?

Reply:
> ok, I'll still update the arguments to put them in the same order (just because)

👍 that'll also make it easier to check what functionality is the same vs different from upstream

> My preference would be to stick lonboard-specific stuff like wireframe and elevation at the end of the signature

Let's remove any Lonboard-specific stuff for now and just get the minimal integration merged. That will be easier and faster to review and merge, and then we can add Lonboard-specific stuff as a follow up